cmd/private-org-sync/main_test.go

```diff
@@ -467,6 +468,71 @@ func TestMirror(t *testing.T) {
+				{call: "init"},
+				{call: "remote get-url org-repo"},
+				{call: "ls-remote --heads org-repo", output: "source-sha refs/heads/branch"},
+				{call: "fetch --tags org-repo branch --depth=2"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "rev-parse --is-shallow-repository", output: "true"},
+				{call: "fetch --tags org-repo branch --depth=4"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "rev-parse --is-shallow-repository", output: "true"},
+				{call: "fetch --tags org-repo branch --depth=8"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "rev-parse --is-shallow-repository", output: "true"},
+				{call: "fetch --tags org-repo branch --depth=16"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "rev-parse --is-shallow-repository", output: "true"},
+				{call: "fetch --tags org-repo branch --depth=32"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "rev-parse --is-shallow-repository", output: "true"},
+				{call: "fetch --tags org-repo branch --depth=64"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "rev-parse --is-shallow-repository", output: "true"},
+				{call: "fetch --tags org-repo branch --unshallow"},
+				{
+					call:     "push --tags --dry-run https://TOKEN@github.com/dest/repo FETCH_HEAD:refs/heads/branch",
+					exitCode: 1,
+					output:   "...Updates were rejected because the remote contains work that you do...",
+				},
+				{call: "remote add dest-repo https://github.com/dest/repo"},
+				{call: "merge org-repo/branch"},
+				{call: "checkout -b branch-priv"},
+				{call: "fetch dest-repo"},
+				{call: "merge dest-repo/branch -m 'Periodic merge from DPTP; pub->priv'"},
+				{call: "push --tags --dry-run https://TOKEN@github.com/dest/repo branch-priv:branch"},
```

Comment:
The testcase illustrates how this can be done even simpler: there's no need to make a new named remote nor a named local branch.

```
git fetch https://TOKEN@github.com/dest/repo branch # fetch destination (and only it's branch, not all of them)
git co FETCH_HEAD                                                     # checks out the destination branch so it can be merged into
git merge org-repo/branch                                             # merge the source branch into the current one
git push https://TOKEN@github.com/dest/repo HEAD:branch # push to destination
```